### PR TITLE
advantage: implement view as functionality

### PR DESCRIPTION
**Pull request by frankban**
## Done

advantage: implement view as functionality.

This will allow admins to view /advantage as a specific user by providing the `email` query.

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Go to http://0.0.0.0:8001/advantage?test_backend=true , log in and check that you see your contracts as usual.
- Now, assuming you have the contract ACL in ua-contracts staging, go to http://0.0.0.0:8001/advantage?test_backend=true&email=allan.vidal@canonical.com and you should see Allan's contracts.

### CODE REVIEW

File: webapp/advantage/views.py [hunk 1]
```diff
@@ -121,9 +123,14 @@ def advantage_view(**kwargs):
                 }
 
         for contract in account["contracts"]:
-            contract_id = contract["contractInfo"]["id"]
-            contract["token"] = advantage.get_contract_token(contract_id)
             contract_info = contract["contractInfo"]
+            if not contract_info.get("items"):
```
Comment (albertkol) [inline webapp/advantage/views.py:127]: I am not sure I understand what this `if` or `TODO` is about.
..and is this work to be done on our end or on your end?

If this is something we have to do, I'd rather have an issue created for it and have the whole `if` removed.

Comment (frankban) [reply webapp/advantage/views.py:127]: Our end. We might have some leftover contracts with no items, and therefore without EffectiveFrom, EffectiveTo etc, leading to key errors in the code below. This check makes the frontend ignore such contracts.

Comment (albertkol) [reply webapp/advantage/views.py:127]: Could we have the API not return contracts with no items instead?

Comment (frankban) [reply webapp/advantage/views.py:127]: there should not be contracts with no items, hence the TODO for me. Once fixed on ua-contracts we can remove this check.

**Pull request merged**